autoload/phpactor.vim

```diff
@@ -435,14 +435,38 @@ endfunction
 " RPC -->-->-->-->-->--
 """""""""""""""""""""""
 
+function! s:searchDirectoryUpwardForRootPatterns(initialDirectory, rootPatterns, fallbackDirectory)
+  let l:directory = a:initialDirectory
+  while index(g:phpactorGlobalRootPatterns, l:directory) < 0
+    for l:pattern in a:rootPatterns
+      if (filereadable(l:directory .'/'. l:pattern))
+        return l:directory
+      endif
+    endfor
+    let l:directory = fnamemodify(l:directory, ':h')
+  endwhile
+
+  if index(g:phpactorGlobalRootPatterns, l:directory) >= 0
+    let l:directory = a:fallbackDirectory
+  endif
```

Comment:
Notice that there these patterns are used also to exclude them.

The one caveat is a case when `a:fallbackDirectory` is also one of them ;) 

What do you think about writing some tests?

Reply:
Yes - testing is relatively easy with https://github.com/junegunn/vader.vim tests are in `tests/VimPlugin` can help out there if needed - there are several examples already. (we don't reallyt have the habbit of writing tests for the plugin, but we should)